src/ray/raylet/worker_killing_policy_group_by_owner.h

```diff
@@ -0,0 +1,95 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <gtest/gtest_prod.h>
+
+#include "absl/container/flat_hash_set.h"
+#include "ray/common/memory_monitor.h"
+#include "ray/raylet/worker.h"
+#include "ray/raylet/worker_killing_policy.h"
+
+namespace ray {
+
+namespace raylet {
+
+/// Key groups on its owner id. For non-retriable task the owner id is itself,
+/// Since non-retriable task forms its own group.
+struct GroupKey {
+  GroupKey(const TaskID &owner_id) : owner_id(owner_id) {}
+  const TaskID &owner_id;
+};
+
+class Group {
```

Comment:
I think it can go both ways - my interpretation of "data object" is a struct may contain only accessor functions (or derivatives of) and debug / basic functions like toString / hash / equals etc

The current group implementation has logic to find the worker to kill, if we move that out of the class and make it a static function, then I think it looks more like a pure data object. I think it is nicer to keep Group as a class and not a pure data object, since there was also comment around making the Group a state object for performance reasons, and I think it is nice to keep the existing encapsulation of the logic for choosing the worker to kill within a group, as opposed to making that a static helper function.

Either way I don't think it is a big issue, so let me know if we need to refactor the code or just see if it is possible to just rename it to a struct, without further code changes